Tests/BowLaws/HashableKLaws.swift

```diff
@@ -0,0 +1,22 @@
+import SwiftCheck
+import Bow
+import BowGenerators
+
+public class HashableKLaws<F: HashableK & ArbitraryK, A: Arbitrary & Hashable> {
+    
+    public static func check() {
+        coherenceWithEquality()
+    }
+    
+    private static func coherenceWithEquality() {
+        property("Equal objects have equal hash") <~ forAll { (fa1: KindOf<F, A>, fa2: KindOf<F, A>) in
+            return (fa1.value == fa2.value) ==> {
+                var hasher1 = Hasher()
+                var hasher2 = Hasher()
+                fa1.value.hash(into: &hasher1)
+                fa2.value.hash(into: &hasher2)
+                return hasher1.finalize() == hasher2.finalize()
+            }
+        }
```

Comment:
I think this property will rarely find a mistake in the implementation, as I guess the chance of generating two equal inputs `fa1` and `fa2` is low, therefore the right hand side of the implication will rarely be invoked.

Reply:
Equal instances are generated ~350 times for the whole Bow test suite, so I guess it's testing something. Might not be the most efficient test, but I think It does no harm to keep it.

Reply:
I think we can use a custom generator to guarantee equal instances are generated in the same proportion as different instances in this particular property. We can make such custom generator as:

```swift
let generator: Gen<(KindOf<F, A>, KindOf<F, A>)> = {
        let equal = KindOf<F, A>.arbitrary.map { fa in (fa, fa) }
        let distinct = Gen.zip(KindOf<F, A>.arbitrary, KindOf<F, A>.arbitrary)
        return Gen.one(of: [equal, distinct])
    }()
```

Then, we can rewrite the property using:

```swift
property("Equal objects have equal hash") <~ forAllNoShrink(self.generator) { (tuple: (KindOf<F, A>, KindOf<F, A>)) in
```

Notice that I am passing the generator and using `forAllNoShrink`, as the tuple does not conform to `Arbitrary`. With this, we are forcing to have 50% chance of getting equal values for this property and strengthening its value.

(It may not be this exact syntax, I haven't tested this code on Xcode).

Reply:
The problem I see with this approach is that we don't test pairs of objects that are different (as in different memory layout) but equal according to Equatable.

For example, if I had this type
```swift
struct DummyPair<A>: Equatable & Hashable {
    let first: A
    let second: A

    static func == (lhs: DummyPair, rhs: DummyPair) -> Bool {
        lhs.first == rhs.first
    }
}
```
the generator you propose would never generate the following case
```swift
DummyPair(1, 2) == DummyPair(1, 3)
// true
DummyPair(1, 2).hashValue == DummyPair(1, 3)
// false, because the default Hashable implementation will take into account the second value
```
and we wouldn't detect that we are missing a custom implementation of Hashable here.

Reply:
I see, what you mean, but isn't that covered in the `distinct` generator? That is the same generator we are using when none is passed explicitly. If you want, we can tweak the probabilities of using each generator (`equal` vs `distinct`) to be 0.3-0.7 instead of 0.5-0.5 or something like that.

Reply:
Ah sorry, you are right, I read too fast. Looks good to me!